tests/chainer_tests/functions_tests/pooling_tests/test_spatial_pyramid_pooling_2d.py

```diff
@@ -40,11 +47,22 @@ def setUp(self):
             -1, 1, (self.n, self.output_dim, 1, 1)).astype(self.dtype)
         self.ggx = numpy.random.uniform(-1, 1, shape).astype(self.dtype)
 
+    def func(self, x):
+        if hasattr(self, 'pooling'):
+            y = functions.spatial_pyramid_pooling_2d(
+                x, self.pyramid_height, pooling=self.pooling)
+        elif hasattr(self, 'pooling_class'):
+            with testing.assert_warns(DeprecationWarning):
+                y = functions.spatial_pyramid_pooling_2d(
+                    x, self.pyramid_height, self.pooling_class)
+        else:
+            raise ValueError('Pooling operation must be specified.')
```

Comment:
Use `assert` instead of exception.

Reply:
I see, considering that test parameters are implementations? Will fix.

Reply:
Fixed.

Reply:
Because it should never happen and it does not make sense to inform the error to the caller.